src/internal/util/logger.ts

```diff
@@ -1,119 +1,286 @@
+        } else {
+            unRedirectConsoleLogging();
+        }
+
+        winstonLogger.silent = false;
+    }
+
+    // Set up file logging
+    if (config.file.enabled) {
+
+        let filename = config.file.filename;
+
+        if (!filename) {
+            const appDir = __dirname.split(p.join("node_modules", "@atomist", "automation-client"))[0];
+            let pj: { name: string; };
+            try {
+                pj = require(p.join(appDir, "package.json"));
+            } catch (e) {
+                pj = { name: "atm-client" };
+            }
+            filename = p.join(
+                os.homedir(),
+                ".atomist",
+                "log",
+                `${pj.name.replace(/^.*\//, "")}-local.log`);
+        }
+
+        const path = p.resolve(filename);
+        fs.mkdirsSync(p.dirname(path));
+
+        const ft = new winston.transports.File({
             filename: p.basename(path),
             dirname: p.dirname(path),
+            level: config.file.level || config.console.level,
             maxsize: 10 * 1024 * 1024,
             maxFiles: 10,
             tailable: true,
-            zippedArchive: true,
-            json: false,
-            formatter,
+            // zippedArchive: true, // see https://github.com/winstonjs/winston/issues/1128
+            format: winston.format.combine(
+                getFormat(config.file.format),
+                winston.format.uncolorize(),
+            ),
         });
+        winstonLogger.add(ft);
+        winstonLogger.silent = false;
+    }
 }
 
-export function formatter(options: any): string {
-    if (LoggingConfig.format === "cli") {
-        return options.message;
+/**
+ * Configure the logging sub-system based on the provided Configuration object
+ * @param configuration
+ */
+export function clientLoggingConfiguration(configuration: Configuration): LoggingConfiguration {
+    const lc: LoggingConfiguration = {
+        ...DefaultClientLoggingConfiguration,
+    };
+    if (configuration.logging) {
+        if (configuration.logging.level) {
+            lc.console.enabled = true,
+                lc.console.level = configuration.logging.level;
+            lc.console.format = LoggingFormat.Full;
+        }
+        if (configuration.logging.file) {
+            if (configuration.logging.file.enabled === true) {
+                let filename = p.join(".", "log", `${configuration.name.replace(/^.*\//, "")}.log`);
+                if (configuration.logging.file.name) {
+                    filename = this.configuration.logging.file.name;
+                }
+                lc.file = {
+                    enabled: true,
+                    level: configuration.logging.file.level || configuration.logging.level,
+                    filename,
+                    format: LoggingFormat.Full,
+                };
+            }
+        }
     }
+    return lc;
+}
 
+const clientFormat = info => {
+    const c = winston.format.colorize();
```

Comment:
So calling the function colorize returns an object that has a function colorize?

Reply:
Yes.

Reply:
Winston now ships types so that is call compile-time checked.